yellow-paper/docs/public-vm/avm-circuit.md

```diff
@@ -0,0 +1,40 @@
+---
+sidebar_position: 1
+---
+
+# AVM Circuit
+
+## Memory
+**VM memory** refers to the VM circuit's top-level memory structure. Since the VM circuit executes an entire public execution request, VM memory contains a dedicated region for each of a request's message calls.
+
+**Call memory** refers to a message call's dedicated memory region. Each call's memory region is of the same size `CALL_MEMORY_SIZE` and starts at offset `callMemoryOffset = callPointer * CALL_MEMORY_SIZE`.
+
+### Protected memory and user memory
+A call's memory is further divided into protected memory and user memory subregions.
+
+**Protected memory** refers to the subsection of a call's memory region that is not explicitly addressable by user code. Dedicated instructions (like `ADDRESS`, or `JUMP`) can read or write protected memory, but when user code performs an explicit memory access like `M[offset]`, it will never touch protected memory.
```

Comment:
@dbanks12 I am not sure whether this is really set into stone on whether protected memory lives in the same address space as user memory or whether we will introduce a special protected memory that we address with the flag.

We should discuss this in one of our VM session. It is also entirely possible that we decide at a later stage. If so, you might describe in this document that there will be 2 distinct memory regions without saying how we address them separately. You might add a note like "OPEN:" It is not yet decided whether a protected and user memory share the same address space (enforcement of boundaries) or they will be completely segregated.

Reply:
As we discussed offline, I'll rework this doc so that protected vs user memory do not share the same top-level address space.